src/server/pfs/server/driver.go

```diff
@@ -209,6 +209,15 @@ func absent(key string) etcd.Cmp {
 }
 
 func (d *driver) createRepo(ctx context.Context, repo *pfs.Repo, description string, update bool) error {
+	// Check that the user is logged in (user doesn't need any access level to
+	// create a repo, but they must be authenticated if auth is active)
+	whoAmI, err := d.pachClient.AuthAPIClient.WhoAmI(auth.In2Out(ctx),
+		&auth.WhoAmIRequest{})
+	authIsActivated := !auth.IsErrNotActivated(err)
+	if authIsActivated && err != nil {
+		return fmt.Errorf("error authenticating (must log in to create a repo): %v",
+			grpcutil.ScrubGRPC(err))
+	}
```

Comment:
I don't think there's a test case for this behavior included. We should have a test for this change.

Reply:
Done (it's not really intended to be a change in semantics, but I think it basically is; this error used to get returned in the STM below)